### ISSUE TITLE
[FR] Allow for pipeline to be defined in a file

### ISSUE THREAD

**Issue by rafamel** [pikapkg/pack#45]
Hi, congrats on the project!

It would be great to be able to define the pack pipeline in a different js file rather than in `package.json`. I'm suggesting something like `pack.config.js` - [liftoff](https://github.com/js-cli/js-liftoff) could simplify implementation. Is that something we might get in the future?

Rafa.

**Comment by nicocevallos** [comment]
I agree with this feature request, I just wanted to ask the same lol

**Comment by stramel** [comment]
Wonder if something like cosmicconfig could be useful here?

**Comment by nicocevallos** [comment]
`cosmicconfig` looks really good. Today I created a little patch to allow to read the JS file suggested by @rafamel, but think would be great to allow other kind of config files, even though I think JS is better because you can change the output using env vars.

What do you think @fredkschott?

**Comment by rafamel** [comment]
Hi guys. I believe moving this forwards would also solve #29, as like @NicoCevallos said, we could manipulate the exported object as desired using env vars -hence the degree of flexibility is higher than what is suggested there.

Regarding implementation, not sure what @FredKSchott would think about it but (and I know it was me who suggested `liftoff`) I'm not too much of a fan of magic, and it does looks like both `liftoff` and `cosmiconfig` are a little bit too magical for my taste. I recently implemented something like this for `poseup`, and getting the file recursing up from a dir using `find-up` would just be:

```javascript
const file = await findUp(
  ['.js', '.json', '.yml', '.yaml'].map((ext) => 'pack.config' + ext),
  { cwd: directory }
);
```

In case there's any interest, this is [the source for the file getter](https://github.com/rafamel/poseup/blob/master/src/builder/get-file.ts), which then just goes through [a switch statement to read it.](https://github.com/rafamel/poseup/blob/master/src/builder/read-file.ts)
